.github/workflows/publish-branch.yaml

```diff
@@ -0,0 +1,40 @@
+name: Publish Branch
+
+on:
+    push:
+        branches:
+            - main
+            - dev
+
+jobs:
+    release_candidate:
+        runs-on: ubuntu-latest
+        env:
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+            NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.head_ref }}
+
+            - name: Setup Project
+              uses: ./.github/actions/setup
+
+            - name: Setup npm credentials file
+              run: echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" >> .npmrc
+
+            - name: Get current package.json version
+              run: echo "PACKAGE_VERSION=$(npm pkg get version | tr -d '"')" >> $GITHUB_ENV
+
+            # get the sort SHA and add it into the environment variables
+            - name: Setup Branch Release Candidate Version
+              run: echo "BRANCH_VERSION=$PACKAGE_VERSION-$BRANCH_NAME.${GITHUB_SHA::7}" >> $GITHUB_ENV
+
+            - name: Bump version
+              run: npm version $BRANCH_VERSION --no-git-tag-version --force
+
+            - name: Publish a new branch release candidate version
+              run: npm publish --tag $BRANCH_NAME
+              env:
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
The `env.NPM_TOKEN` is defined at the top-level of this job, you don't need to define it again here